### ISSUE TITLE
[BUG]: （Steam Deck Version） PCSX2 Shut Down Issue

### ISSUE THREAD

**Issue by ChenranXiyu** [PCSX2/pcsx2#10769]
### Describe the Bug

Steam Deck Verision ( pcsx2-v1.7.5540-linux-flatpak-x64-Qt.flatpak ) . Whether in DESKTOP or SteamOS, when I click "Shut Down" , it's supposed to just exit the ROM, but the main PCSX2 program actually shuts right down.

### Reproduction Steps

Do what I describe in Steam Deck Verision（pcsx2-v1.7.5540-linux-flatpak-x64-Qt.flatpak）

### Expected Behavior

Plz fix it out, Thx!

### PCSX2 Revision

v1.7.5540

### Operating System

Linux (64bit) - Specify distro below

### If Linux - Specify Distro

KDE( Steam Deck )

**Comment by weirdbeardgame** [comment]
It appears to be an issue with Flatapak builds, the AppImage seems alright.

**Comment by xiota** [comment]
I see same behavior on Arch Linux when using Vulkan.  When I switch to OpenGL, `pcsx2` returns to game list, as expected.

In terminal, following is displayed:
```
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/13.2.1/../../../../include/c++/13.2.1/bits/atomic_base.h:477: void std::__atomic_base<bool>::store(__int_type, memory_order) [_ITp = bool]: Assertion '__b != memory_order_acquire' failed.
Aborted (core dumped)
```

From backtrace with `gdb`, seems related to `libc.so.6`.  Seems to be related to commit 9809265be4cf61c5406a79642254b7cbbcb00ee1.  I have not yet tried building with that commit reverted.

**Comment by kamfretoz** [comment]
Fixed with #10775
